agent/embeddedconsul.go

```diff
@@ -0,0 +1,113 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+
+	consulAgent "github.com/hashicorp/consul/agent"
+	configAgent "github.com/hashicorp/consul/agent/config"
+	"github.com/hashicorp/consul/agent/connect"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/hashicorp/go-uuid"
+)
+
+func NodeID() string {
+	id, err := uuid.GenerateUUID()
```

Comment:
I don't think we should do this. At least in my head, the consul NodeID should be permanent, We will end up using it to store the data related to the nodes in the KV storage.

Maybe we should some machine specific values to create this UUID